### ISSUE TITLE
Wrong spacegrep executable installed by pip (was: Adding support for CFML)

### ISSUE THREAD

**Issue by reillybr** [semgrep/semgrep#1937]
**Describe the solution you'd like**

It would be great to have support for Coldfusion/CFML in semgrep.  (Edit:  I'm looking at #1392 as well for some guidance on trying to take a shot at adding it.)

I can successfully run basic rules on CFML code in the web Live Editor, but it doesn't seem to work on the command line.  

Error output:

% **semgrep -f test.yaml vuln.cfm**
running 1 rules...
spacegrep output could not be parsed as JSON.
Traceback (most recent call last):
  File "/usr/local/bin/semgrep", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/semgrep/__main__.py", line 17, in main
    cli()
  File "/usr/local/lib/python3.8/dist-packages/semgrep/cli.py", line 378, in cli
    semgrep.semgrep_main.main(
  File "/usr/local/lib/python3.8/dist-packages/semgrep/semgrep_main.py", line 223, in main
    rule_matches_by_rule, debug_steps_by_rule, semgrep_errors, num_targets = CoreRunner(
  File "/usr/local/lib/python3.8/dist-packages/semgrep/core_runner.py", line 531, in invoke_semgrep
    findings_by_rule, debug_steps_by_rule, errors, num_targets = self._run_rules(
  File "/usr/local/lib/python3.8/dist-packages/semgrep/core_runner.py", line 499, in _run_rules
    rule_matches, debugging_steps, errors, rule_targets = self._run_rule(
  File "/usr/local/lib/python3.8/dist-packages/semgrep/core_runner.py", line 390, in _run_rule
    output_json = run_spacegrep(patterns, targets)
  File "/usr/local/lib/python3.8/dist-packages/semgrep/spacegrep.py", line 49, in run_spacegrep
    errors.extend(output_json["errors"])
KeyError: 'errors'

--
SAMPLE CODE (vuln.cfm):

```
<h3>cfexecute</h3>
<!--- This executes a command and outputs it to a file --->

<cfexecute name = "#form.command#"
arguments = "#form.args#"
outputFile = "C:\Temp\output.txt"
timeout = "1">

<!--- This reads and outputs the entire file contents. --->
<cfscript>
myfile = FileRead(#url.file#);
WriteOutput("#myfile#");
</cfscript>
```

SAMPLE RULES FILE (test.yaml):

```
rules:
- id: dangerous-function
  patterns:
  - pattern-either:
    - pattern-inside: '<cfexecute ...>'
    - pattern-inside: '<cfinclude ...>'
    - pattern-inside: 'FileRead(...)'
  - pattern-either:
    - pattern: '#url...#'
    - pattern: '#form...#'
    - pattern: '#cookie...#'
  languages:
  - generic
  severity: WARNING
  message: User input in dangerous function
```

**Comment by dlukeomalley** [comment]
Thanks for filing this issues @otbar! It looks like there are two separate issues here, one with the "generic" language and a second request for adding Coldfusion/CFML. 

@mjambon Thoughts on the "generic" bug? Do you want a separate ticket to track it?

@aryx Any guidance on adding new language support?

**Comment by aryx** [comment]
Regarding language support, @colleend is working on adding support for Kotlin this month, and while doing so she will try to write some docs to help other people!

**Comment by ievans** [comment]
@nbrahms possibly the same issue as https://github.com/returntocorp/semgrep/issues/1949?

**Comment by mjambon** [comment]
@otbar we fixed the packaging bug (#1949, #1973) that was causing the cryptic error message. It will be available next week when we release semgrep 0.31.0. Thanks for your patience!
